File: app/src/main/java/me/tatarka/betterfragment/sample/NavigationFragment.java
package me.tatarka.betterfragment.sample;

import android.os.Bundle;
import android.view.View;

import androidx.annotation.Nullable;
import androidx.appcompat.widget.Toolbar;
import androidx.navigation.NavController;
import androidx.navigation.ui.NavigationUI;
import me.tatarka.betterfragment.Fragment;

import static androidx.navigation.Navigation.findNavController;

public class NavigationFragment extends Fragment {
    @Override
    public void onCreate(@Nullable Bundle savedState) {
        super.onCreate(savedState);
        setContentView(R.layout.navigation);
        final NavController controller = findNavController(findViewById(R.id.nav));
        Toolbar toolbar = findViewById(R.id.toolbar);
        NavigationUI.setupWithNavController(toolbar, controller);
        findViewById(R.id.root).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                controller.navigate(R.id.root);
            }
        });
        findViewById(R.id.dest1).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                controller.navigate(R.id.dest1);
            }
        });
        findViewById(R.id.dest2).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                controller.navigate(R.id.dest2);
            }
        });
    }
}

File: app/src/main/java/me/tatarka/betterfragment/sample/NavigationFragment.kt
package me.tatarka.betterfragment.sample

import android.os.Bundle
import android.view.View

import javax.inject.Inject
import androidx.appcompat.widget.Toolbar
import androidx.navigation.NavController
import androidx.navigation.Navigation
import androidx.navigation.ui.NavigationUI
import me.tatarka.betterfragment.app.Fragment

import androidx.navigation.Navigation.findNavController

class NavigationFragment @Inject constructor() : Fragment(), NavInterface {

    lateinit var controller: NavController

    override fun onCreate(savedState: Bundle?) {
        super.onCreate(savedState)
        setContentView(R.layout.navigation)
        controller = findNavController(requireViewById(R.id.nav))
        NavigationUI.setupWithNavController(requireViewById<Toolbar>(R.id.toolbar), controller)
        requireViewById<View>(R.id.root).setOnClickListener { controller.navigate(R.id.root) }
        requireViewById<View>(R.id.dest1).setOnClickListener { controller.navigate(R.id.dest1) }
        requireViewById<View>(R.id.dest2).setOnClickListener { controller.navigate(R.id.dest2) }
    }

    override fun onBackPressed(): Boolean = controller.popBackStack()

    override fun onNavigateUp(): Boolean = controller.navigateUp()
}
